theme/moth.js

```diff
@@ -110,6 +105,90 @@ function showPuzzles(teamId) {
   document.getElementById("puzzles").appendChild(spinner)
   heartbeat(teamId)
   setInterval(e => { heartbeat(teamId) }, 40000)
+  drawCacheButton(teamId)
+}
+
+function drawCacheButton(teamId) {
+  let cacher = document.createElement("li");
+  let cache_button = document.createElement("a");
+  cache_button.innerText = "Cache";
+  cache_button.title = "Cache an offine copy of current content";
+  cache_button.href = "#";
+  cache_button.addEventListener("click", async function() {
+    toast("Caching all currently-open content");
+    await fetchAll(teamId);
+    toast("Done caching content");
+  });
+  cacher.appendChild(cache_button);
+  document.getElementsByTagName("nav")[0].getElementsByTagName("ul")[0].appendChild(cacher);
+
+  function updateCacheButton() {
+    let headers = new Headers();
+    headers.append("pragma", "no-cache");
+    headers.append("cache-control", "no-cache");
+    fetch("current_manifest.json?id=" + teamId, {method: "HEAD", headers: headers})
```

Comment:
We should look at that outside of this PR, since it'd require more server-side changes that would affect unrelated functionality. This was just the hammer method for really making sure that the browser didn't try to use a cached result when we wanted fresh content